frontend/src/components/wallet/Disconnect.vue

```diff
@@ -0,0 +1,18 @@
+<template>
+  <slot data-test="disconnect-trigger" v-bind="{ disconnect }"> </slot>
```

Comment:
Hm, I think this is maybe split in a too complex way. I don't see why a function from a composable should be passed to its parent via the slot props in this context. This component does not do anything else than passing the function to the parent. The parent could just call the composable itself. A component not adding any actual HTML also looks wrong to me.

I think the right approach here would be to just move the `span` element, which is in the `Disconnect` component in the parent at the moment, to the `Disconnect` component directly (see my other comment). As a result you would not need to pass the disconnect function up to the parent.

Reply:
> The parent could just call the composable itself

I think composables are a way to make code reusable, so it is a good idea to use them in multiple places. In the best case their interface / parameters just don't change. Anyway, I don't prefer this approach here.

> I think the right approach here would be to just move the span element

Hm, I would say just write another component that renders it as a button. Use the composable there, as we have it for reusing logic.
You could also just use the `button` tag anyway here and style without the look of a button. Would actually make sense semantically. And then let the parent component add styling if we really need it in multiple places.
Moving the `span` (or `button` if you like to change it) into the `Disconnect`  component is still my preference.

Having the logic split among multiple components feels not easily understandable to me. There is even a name for the pattern you used here. [Renderless Components](https://vuejs.org/guide/reusability/composables.html#vs-renderless-components). Only thing is that you mixed it with a composable. In the link, the vue documentation also not recommends this pattern:
> The recommendation is to use composables when reusing pure logic, and use components when reusing both logic and visual layout.

Reply:
Alright, switched to the suggested approach. Thanks for taking the time to elaborate, this was a valuable discussion for me.